File: src/main/java/com/ggardet/doctor/spi/dto/DoctorDTO.java
package com.ggardet.doctor.spi.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;

import java.util.UUID;

public record DoctorDTO(
        UUID id,

        @NotBlank String firstname,

        @NotBlank String lastname,

        @NotBlank String specialty,

        String licenseNumber,

        String phoneNumber,

        String email,

        UUID hospitalId,

        @Positive Integer yearsExperience
) {
}

File: src/main/kotlin/com/ggardet/doctor/spi/dto/DoctorDTO.kt
package com.ggardet.doctor.spi.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive
import java.util.*

data class DoctorDTO(
    val id: UUID?,
    
    @field:NotBlank
    val firstname: String,
    
    @field:NotBlank
    val lastname: String,
    
    @field:NotBlank
    val specialty: String,
    
    val licenseNumber: String?,
    
    val phoneNumber: String?,
    
    val email: String?,
    
    val hospitalId: UUID?,
    
    @field:Positive
    val yearsExperience: Int?
)